### PR TITLE
AI-144: Moved vacancies style from shoreditch to civihr

### PR DESCRIPTION
**Pull request by mukeshcompucorp**
## Overview
- This PR moves style for vacancies module to civihr from shoreditch with code cleanups.
- There are no visible changes except one (showing button pane on popup as per screenshots below)

## Before
before-2

## After
after-2

## Technical Details
1. Adds code to call civihr css of form pages
`function bootstrapcivihr_civicrm_buildForm($formName, &$form) {
    CRM_Core_Resources::singleton()->addStyleFile('org.civicrm.bootstrapcivihr', 'css/civihr.css');
}`

2. Excluded page related to css from adding prefix #bootsrap-theme in gulpfile.js
`exclude: [/^html/, /^body/, /page-civi/]` 

3. Added common file common/placeholders.css to make it easy finding and editing

4. It divides page /org.civicrm.shoreditch/scss/civicrm/vacancies/_vacancies.scss into 3 partials based on it's actual pages
4.1. civihr/org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss
4.2. civihr/org.civicrm.bootstrapcivihr/scss/pages/case.scss
4.3. civihr/org.civicrm.bootstrapcivihr/scss/pages/activity.scss

5. civihr/org.civicrm.bootstrapcivihr/scss/pages/reports.scss file is changed to provide proper page class instead of `body[class*="page-civihr-reports"]`

## Comments
- This PR does not focus on creating any new styles. it just cleans css and moves it to civihr

---

- [x] Tests Pass

### CODE REVIEW

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                border-left: 0 !important;
+                width: auto;
+                display: none;
+
+                &:first-child {
+                  display: block;
+                }
+
+                a {
+                  font-weight: 600;
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: Add font-weight to a variable

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+                  color: $crm-white;
+
+                  .edit-icon {
+                    background-image: none;
+                    text-indent: 0;
+                    top: -3px;
+                    margin: 0;
+
+                    &:after {
+                      @include fa-icon($font-size-base, '\f040', $crm-white);
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: Add icon to a variable

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+              }
+
+              h3 {
+                background-color: $crm-copy;
+                padding: 8px 15px;
+
+                > a {
+                  color: $crm-white;
+
+                  .edit-icon {
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: You have many nested levels. please try to reduce the hierarchy levels to 3/4 as a max

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+
+        > table {
+          width: 100%;
+          background-color: $crm-white;
+          border: 0;
+
+          tr {
+            &:nth-child(2) {
+              td {
+                line-height: 25px;
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: I recommend using em instead of px for line-height

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+    .crm-accordion-header h2 {
+      color: $gray-darker !important;
+      display: inline-block;
+    }
+
+    .hr-vacancy {
+      width: 49%;
+      margin-bottom: 20px;
+      border: 1px solid $crm-background;
+      border-top-left-radius: 3px;
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: You can use variables for radius

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+        border-bottom: 0;
+      }
+    }
+
+    .advmultiselect tr:first-child td:nth-child(2) {
+      vertical-align: top;
+
+      input[type='button']:first-child {
+        margin-left: 0 !important;
+        margin-bottom: 15px !important;
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: I see many !important in this PR. is it all necessary? Can u try to reduce the !important as much as you can?

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 7]
```diff
@@ -0,0 +1,315 @@
...
+      background: $crm-blue-light;
+    }
+
+    > .add-icon {
+      background-image: none;
+      text-indent: 0;
+      margin-right: 3px;
+
+      &:after {
+        @include fa-icon($font-size-base, '\f067',$crm-white);
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: Use variable for icons

File: org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss [hunk 8]
```diff
@@ -0,0 +1,315 @@
...
+        min-width: 200px !important;
+      }
+
+      .crm-form-multiselect {
+        min-width: 230px;
+        width: 230px;
+      }
+    }
+
+    @media (max-width: 1100px) and (min-width: 861px) {
```
Comment (meladawy) [inline org.civicrm.bootstrapcivihr/scss/pages/vacancies.scss]: Can't we use bootstrap screen breakpoints variables instead?

File: org.civicrm.bootstrapcivihr/bootstrapcivihr.php [hunk 9]
```diff
@@ -117,3 +117,9 @@ function bootstrapcivihr_civicrm_pageRun($page) {
     CRM_Core_Resources::singleton()->addStyleFile('org.civicrm.bootstrapcivihr', 'css/civihr.css');
   }
 }
+/**
+ * Implements hook_civicrm_buildForm().
+ */
+function bootstrapcivihr_civicrm_buildForm($formName, &$form) {
+  CRM_Core_Resources::singleton()->addStyleFile('org.civicrm.bootstrapcivihr', 'css/civihr.css');
```
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/bootstrapcivihr.php]: why is this needed? I thought that the `_pageRun` hook would add the css in any page?

File: org.civicrm.bootstrapcivihr/scss/common/placeholders.scss [hunk 10]
```diff
@@ -0,0 +1,25 @@
+%search-filter-block {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/scss/common/placeholders.scss]: any scss partial needs to start with `_` (`_placeholders.scss`)

File: org.civicrm.bootstrapcivihr/scss/overrides/_variables.scss [hunk 11]
```diff
@@ -1,6 +1,8 @@
 @import "SCSSROOT/org.civicrm.shoreditch/scss/bootstrap/overrides/variables";
 @import "SCSSROOT/org.civicrm.shoreditch/base/scss/overrides/variables";
 @import "SCSSROOT/org.civicrm.shoreditch/base/scss/vendor/bootstrap/variables";
+@import "SCSSROOT/org.civicrm.shoreditch//scss/bootstrap/vendor/font-awesome/variables";
```
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/scss/overrides/_variables.scss]: you have `//`

File: org.civicrm.bootstrapcivihr/scss/overrides/_variables.scss [hunk 12]
```diff
@@ -12,3 +14,4 @@ $steps-height: 24px;
 $steps-compact-height: 28px;
 
 $steps-gutter: ($steps-height / 4);
+$button-dark-color: #AAAAAA;
```
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/scss/overrides/_variables.scss]: do not add new colors, use the one already provided in shoreditch

File: org.civicrm.bootstrapcivihr/scss/pages/reports.scss [hunk 13]
```diff
@@ -1,4 +1,4 @@
-body[class*="page-civihr-reports"] {
+.page-civihr-reports {
 
     .form-inline {
```
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/scss/pages/reports.scss]: fix the indentation

File: org.civicrm.bootstrapcivihr/scss/common/placeholders.scss (hunk 10 above)
Comment (AkA84) [inline org.civicrm.bootstrapcivihr/scss/common/placeholders.scss]: also create a `placeholders/` folder and put each placeholder in a different partial

File: org.civicrm.bootstrapcivihr/bootstrapcivihr.php (hunk 9 above)
Comment (mukeshcompucorp) [reply org.civicrm.bootstrapcivihr/bootstrapcivihr.php]: Yeah but, that's true for only for CiviCRM page classified as _page_. This does not affect on page type _form_
You can check more on that at, https://docs.civicrm.org/dev/en/stable/hooks/hook_civicrm_pageRun/

File: org.civicrm.bootstrapcivihr/scss/pages/reports.scss (hunk 13 above)
Comment (mukeshcompucorp) [inline org.civicrm.bootstrapcivihr/scss/pages/reports.scss]: @AkA84 If this is not usable anymore then, we will remove this file as we discussed.
